packages/clerk-js/src/v4/UserProfile/MfaPage.tsx

```diff
@@ -1,129 +1,88 @@
     }
+  }, []);
+
+  if (card.error) {
+    return <ContentPage.Root headerTitle={title} />;
+  }
+
+  // If only phone_code is available, just render that
+  // Otherwise check if selected
+  if (
+    (secondFactorsAvailableToAdd.length === 1 && secondFactorsAvailableToAdd[0] === 'phone_code') ||
+    selectedMethod === 'phone_code'
+  ) {
+    return <MfaPhoneCodePage />;
+  }
 
-    card.setLoading(phone.id);
-    phone
-      .setReservedForSecondFactor({ reserved: true })
-      .then(() => {
-        card.setIdle();
-        onSuccess();
-      })
-      .catch(err => handleError(err, [], card.setError));
-  };
+  // If only totp is available, just render that
+  // Otherwise check if selected
+  if (
+    (secondFactorsAvailableToAdd.length === 1 && secondFactorsAvailableToAdd[0] === 'totp') ||
+    selectedMethod === 'totp'
+  ) {
+    return <MfaTOTPPage />;
+  }
 
   return (
     <ContentPage.Root headerTitle={title}>
-      <Text>
-        {availableMethods.length
-          ? 'Select a phone number to register for SMS code two-step authentication.'
-          : 'There are no available methods.'}
-      </Text>
-      <Col gap={2}>
-        {availableMethods.map(phone => {
-          const formattedPhone = stringToFormattedPhoneString(phone.phoneNumber);
-          const flag = getFlagEmojiFromCountryIso(parsePhoneString(phone.phoneNumber).iso);
+      <Col gap={4}>
+        <Text>Select a method to add.</Text>
 
-          return (
-            <ArrowBlockButton
-              // elementDescriptor={descriptors.socialButtonsButtonBlock}
-              // elementId={descriptors.socialButtonsButtonBlock.setId(strategy)}
-              // textElementDescriptor={descriptors.socialButtonsButtonBlockText}
-              // textElementId={descriptors.socialButtonsButtonBlockText.setId(strategy)}
-              // arrowElementDescriptor={descriptors.socialButtonsButtonBlockArrow}
-              // arrowElementId={descriptors.socialButtonsButtonBlockArrow.setId(strategy)}
-              key={phone.id}
-              // id={strategyToDisplayData[strategy].id}
-              onClick={() => enableMfa(phone)}
-              isLoading={card.loadingMetadata === phone.id}
-              isDisabled={card.isLoading}
-              icon={
-                <Text
-                  as='span'
-                  sx={theme => ({ fontSize: theme.fontSizes.$sm })}
-                >
-                  {flag}
-                </Text>
-              }
-            >
-              {formattedPhone}
-            </ArrowBlockButton>
-          );
-        })}
-        <AddBlockButton
-          block={false}
-          onClick={onAddPhoneClick}
+        <Grid
+          columns={3}
+          gap={2}
         >
-          Add a phone number
-        </AddBlockButton>
+          <TileButton
+            icon={AuthenticatorApp}
+            onClick={() => setSelectedMethod('totp')}
+            sx={{ height: '96px' }}
```

Comment:
❓ Why are we adding the `height` via `sx`? Can we set it inside TileButton directly? 

Also let's change the above to
```suggestion
            sx={t => ({ height: t.sizes.$24 })}
```

so it can be themed correctly when the user passes in  a custom `appearance.variables` config

Reply:
Moving to `TileButton`